rustls/src/msgs/ffdhe_groups.rs

```diff
@@ -0,0 +1,267 @@
+use crate::NamedGroup;
+
+#[derive(Clone, Copy, PartialEq, Eq)]
+/// Parameters of an FFDHE group, with Big-endian byte order
+pub struct FfdheGroup<'a> {
+    pub p: &'a [u8],
+    pub g: &'a [u8],
+}
+
+impl FfdheGroup<'static> {
+    /// Return the `FfdheGroup` corresponding to the provided `NamedGroup`
+    /// if it is indeed an FFDHE group
+    pub fn from_named_group(named_group: NamedGroup) -> Option<Self> {
+        match named_group {
+            NamedGroup::FFDHE2048 => Some(FFDHE2048),
+            NamedGroup::FFDHE3072 => Some(FFDHE3072),
+            NamedGroup::FFDHE4096 => Some(FFDHE4096),
+            NamedGroup::FFDHE6144 => Some(FFDHE6144),
+            NamedGroup::FFDHE8192 => Some(FFDHE8192),
+            _ => None,
+        }
+    }
+}
+
+impl<'a> FfdheGroup<'a> {
+    /// Return the `NamedGroup` for the `FfdheGroup` if it represents one.
+    pub fn named_group(&self) -> Option<NamedGroup> {
+        match *self {
+            FFDHE2048 => Some(NamedGroup::FFDHE2048),
+            FFDHE3072 => Some(NamedGroup::FFDHE3072),
+            FFDHE4096 => Some(NamedGroup::FFDHE4096),
+            FFDHE6144 => Some(NamedGroup::FFDHE6144),
+            FFDHE8192 => Some(NamedGroup::FFDHE8192),
+            _ => None,
+        }
+    }
+}
+
```

Comment:
Suggest to add comments for this and following constants.
```suggestion

/// FFDHE2048 group defined in [RFC 7919 Appendix A.1]
///
/// [RFC 7919 Appendix A.1]: https://datatracker.ietf.org/doc/html/rfc7919#appendix-A.1
```

Reply:
Good suggestion! Will do.